### PR TITLE
Add section for cranelift

### PR DESCRIPTION
**Pull request by janhohenheim**
Adopts https://github.com/bevyengine/bevy-website/pull/801 by @MalekiRe and changes some text based on feedback by @bjorn3 

~~Blocked by~~:
- [x] https://github.com/rust-lang/rustc_codegen_cranelift/issues/1504
- [ ] https://github.com/rust-lang/rustc_codegen_cranelift/issues/1507
- [ ] https://github.com/rust-lang/rust/pull/127177

Edit: leaving it as a primary Linux recommendation for now.

### CODE REVIEW

File: content/learn/quick-start/getting-started/setup.md
Comment (BD103) [inline content/learn/quick-start/getting-started/setup.md:263]:
```suggestion
# Replace `my_program` with the name of your binary.
[profile.dev.package.my_program]
```

File: content/learn/quick-start/getting-started/setup.md
Comment (BD103) [inline content/learn/quick-start/getting-started/setup.md:267]:
```suggestion
This enables faster compiles for your binary, but builds Bevy and other dependencies with the more-optimized LLVM backend. See the [cranelift setup guide](https://github.com/rust-lang/rustc_codegen_cranelift?tab=readme-ov-file#download-using-rustup) for
```

**Review by BD103 — APPROVED** [review]
This looks very good right now, but one nit I have is to only compile your binary with Cranelift. In the [October 2023 progress report](https://bjorn3.github.io/2023/10/31/progress-report-oct-2023.html), it mentions how you can enable Cranelift for specific packages.

In this case, I think it would be best to recommend to only compile your binary with Cranelift, so that Bevy and the rest still receive better optimizations. This is the same reason why we recommend `opt-level = 3` for dependencies.

**Comment by janhohenheim** [comment]
@BD103 that makes sense, thanks for the suggestion :)

File: content/learn/quick-start/getting-started/setup.md
Comment (bjorn3) [inline content/learn/quick-start/getting-started/setup.md:265]: You can also use

```toml
[profile.dev]
codegen-backend = "cranelift"

[profile.dev.package."*"]
codegen-backend = "llvm"
```

to compile everything with llvm, but compile the main program and all other workspace members with cranelift.

File: content/learn/quick-start/getting-started/setup.md
Comment (bjorn3) [inline content/learn/quick-start/getting-started/setup.md:270]: I don't have time to work on this today, but this bug shouldn't be too hard to fix. I did prefer it gets fixed before merging this PR.

File: content/learn/quick-start/getting-started/setup.md
Comment (bjorn3) [inline content/learn/quick-start/getting-started/setup.md:268]:
```suggestion
This enables faster compiles for your binary, but builds Bevy and other dependencies with the more-optimized LLVM backend. See the [cranelift setup guide](https://github.com/rust-lang/rustc_codegen_cranelift#download-using-rustup) for
```
may be a bit cleaner. Most people won't notice it though.

File: content/learn/quick-start/getting-started/setup.md
Comment (janhohenheim) [reply content/learn/quick-start/getting-started/setup.md]: I like your version more, thanks! [`5f9b4e6` (#1523)](https://github.com/bevyengine/bevy-website/pull/1523/commits/5f9b4e6c7e3bec8fa631ad71a71abb14a94adbe2)

**Comment by janhohenheim** [comment]
Blocked by https://github.com/rust-lang/rustc_codegen_cranelift/issues/1504

**Comment by janhohenheim** [comment]
While we are on the subject, @bjorn3, is there a way to compile my dependencies with LLVM and use cranelift only for my binary on Windows? The way I've set it up locally, I need to run `cargo clif` manually, and I don't know how I would tell it to compile in the specified way. There's no way to do a similar setup to Linux and macOS by just changing one's `config.toml` on Windows, right?

Also, I heard Bevy stuff on Wasm didn't build or didn't run with cranelift. I don't have the time to verify this. Is this still the case? If so, I can add it to the readme as well.

**Comment by bjorn3** [comment]
> is there a way to compile my dependencies with LLVM and use cranelift only for my binary on Windows?

Not really. The codegen-backends option in .cargo/config.toml only works for codegen backend shipped with rustc. Rustc itself supports passing the path to a codegen backend dylib (which cargo-clif internally uses), but for cargo I was asked to restrict this unfortunately.

> Also, I heard Bevy stuff on Wasm didn't build or didn't run with cranelift. I don't have the time to verify this. Is this still the case? If so, I can point it out in the readme as well.

This is still the case.

**Comment by bjorn3** [comment]
https://github.com/rust-lang/rust/pull/127177 would also be really nice to land first. This will ship cg_clif for arm64 macOS.

**Comment by janhohenheim** [comment]
@bjorn3 I've added both links to the PR description for visibility :)

**Comment by janhohenheim** [comment]
@bjorn3 since we discovered a runtime crash on macOS in the meantime, would be fine if we changed the section to recommend cranelift specifically for Linux for now? I can set up a tracking issue and change it to mention macOS at a later point, if that's okay with you.

**Comment by bjorn3** [comment]
Yeah, that is fine. Maybe also explicitly mention why it shouldn't be used on Windows and macOS to avoid people trying to do it anyway.

**Review by bjorn3 — APPROVED** [review]

**Pull request merged**
